Beta_Integrations/integration-ThreatQ.yml

```diff
@@ -70,30 +70,72 @@ script:
 
     PROXY = load_proxy()
 
+    def get_errors_string_from_bad_request(bad_request_results):
+        errors_list = bad_request_results.json().get("errors", [])
+        errors_string = ""
+        error_num = 1
+        if errors_list:
+            errors_string = "Errors from server: \n"
+            for error in errors_list:
+                errors_string += "Error #{0}: {1}\n".format(error_num, error)
+                error_num += 1
+        return errors_string
+
     # ThreatQ auth based on OAuth 2.0 credential grand method
     def tq_access():
         data = {'grant_type': 'password','email': EMAIL, 'password': PASSWORD, 'client_id': CLIENT_ID}
         access_token_response = requests.post(API_TOKEN_URL, data=data, verify=False, allow_redirects=False)
+
         tokens = json.loads(access_token_response.text)
+        if int(access_token_response.status_code) >= 400:
+            errors_string = get_errors_string_from_bad_request(access_token_response)
+            error_message = "Authentication failed, unable to retrieve an access token.\n {}".format(errors_string)
```

Comment:
this should be done as part of `get_errors_string_from_bad_request`

Reply:
I'm not sure I agree.
The "get_errors_string_from_bad_request" function is generic, and in case it will be needed to clarify any other error messages it could be used in other cases too.
This message is specific to authentication, which means it should be a part of tq_access and not the generic command.
What do you think?

Reply:
I don't agree for 100% but ok